File: Código/app/app/src/main/java/pt/ipc/estgoh/ezshop/data/model/Item.java
package pt.ipc.estgoh.ezshop.data.model;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class Item implements Serializable {
    private long id;
    private String name;
    @SerializedName("category_id")
    private long category;

    public Item() {
    }

    public Item(final String aName, final long aCategory) {
        this.name = aName;
        this.category = aCategory;
    }

    public void setName(final String aName) {
        this.name = aName;
    }

    public String getName() {
        return this.name;
    }

    public void setCategory(final long aCategory) {
        this.category = aCategory;
    }

    public long getCategory() {
        return this.category;
    }

    @Override
    public String toString() {
        return "Item{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", category=" + category +
                '}';
    }
}

File: Código/backend/src/main/kotlin/pt/ipc/estgoh/francisco_luis/data/Item.kt
package pt.ipc.estgoh.francisco_luis.data

import kotlinx.serialization.Serializable
import org.jetbrains.exposed.dao.IntEntity
import org.jetbrains.exposed.dao.IntEntityClass
import org.jetbrains.exposed.dao.id.EntityID
import org.jetbrains.exposed.dao.id.IntIdTable

//Representation of the database table
object Items : IntIdTable() {
    val name = varchar("name", 50).uniqueIndex()
    val categoryId = integer("category_id").references(Categories.id)
}


class ItemEntity(id: EntityID<Int>) : IntEntity(id) {
    companion object : IntEntityClass<ItemEntity>(Items)

    var name by Items.name
    var category by Items.categoryId

    override fun toString(): String = "Item($name, $category)"

    fun toItem() = Item(id.value, name, category)
}

@Serializable
data class Item(
    //id is not mandatory
    val id: Int? = null,
    val name: String,
    val category_id: Int? = null,
)